pkg/eventpersister/persister/persister.go

```diff
@@ -310,6 +300,16 @@ func (p *Persister) send(messages map[string]*puller.Message) {
 				continue
 			}
 			evs[id] = eventJSON
+			if err := p.upsertEvaluationCount(event, environmentNamespace); err != nil {
```

Comment:
I moved it to the end because if the upsert in the Bigtable fails, it will duplicate the same event in the Redis until it succeeds.